homeassistant/components/ituran/__init__.py

```diff
@@ -0,0 +1,54 @@
+"""The Ituran integration."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+from .coordinator import IturanDataUpdateCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.DEVICE_TRACKER,
+]
+
+type IturanConfigEntry = ConfigEntry[IturanData]
+
+
+@dataclass
+class IturanData:
+    """Class to store Ituran runtime data."""
+
+    coordinator: IturanDataUpdateCoordinator
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: IturanConfigEntry) -> bool:
+    """Set up Ituran from a config entry."""
+
+    coordinator = IturanDataUpdateCoordinator(hass, entry=entry)
+    await coordinator.async_config_entry_first_refresh()
+    entry.runtime_data = IturanData(coordinator)
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    # Clean up vehicles which are not assigned to the account anymore
+    account_vehicles = {(DOMAIN, license_plate) for license_plate in coordinator.data}
+    device_registry = dr.async_get(hass)
+    device_entries = dr.async_entries_for_config_entry(
+        device_registry, config_entry_id=entry.entry_id
+    )
+    for device in device_entries:
+        if not device.identifiers.intersection(account_vehicles):
+            device_registry.async_update_device(
+                device.id, remove_config_entry_id=entry.entry_id
+            )
```

Comment:
I like this, but should we maybe do this on runtime instead of only on boot? It's a little more effort, but it will be great in the end